src/irc/ClientPool.ts

```diff
@@ -500,17 +487,11 @@ export class ClientPool {
 
     public getNickUserIdMappingForChannel(server: IrcServer, channel: string): {[nick: string]: string} {
         const nickUserIdMap: {[nick: string]: string} = {};
-        const cliSet = this.virtualClients[server.domain].userIds;
-        Object.keys(cliSet).filter((userId: string) => {
-            if (!userId) {
-                return false;
+        for (const [userId, client] of this.virtualClients[server.domain].userIds.entries()) {
```

Comment:
Why is it safe to assume that `this.virtualClients[server.domain]` is defined?

Reply:
I think we assume that if `server` is defined, then it will have a client pool. I think in this case it's a harmless assumption.